packages/manager/modules/vps/src/dashboard/modal/snapshot-download/snapshot-download.controller.js

```diff
@@ -0,0 +1,109 @@
+import 'moment';
+
+export default class SnapshotDownloadController {
+  /* @ngInject */
+  constructor($translate, atInternet, VpsService) {
+    this.$translate = $translate;
+    this.atInternet = atInternet;
+    this.VpsService = VpsService;
+  }
+
+  $onInit() {
+    this.size = null;
+    this.creationDate = moment(this.tabSummary.snapshot.creationDate).format(
+      'LLL',
+    );
+    this.linkExpirationDate = this.getLinkExpirationDate();
+    this.downloadLink = null;
+    this.downloadLinkCurlCommand = null;
+    this.isLoading = false;
+    this.notificationMessage = null;
+    this.errorMessage = null;
+    this.curlDocumentationLink = 'https://curl.se/docs/manual.html';
+  }
+
+  trackClick(hit) {
+    this.atInternet.trackClick({
+      name: `vps::detail::dashboard::snapshot-download::${hit}`,
+      type: 'navigation',
+    });
+  }
+
+  async generateSnapshotDownloadLink() {
+    this.errorMessage = null;
+    this.isLoading = true;
+    this.VpsService.getSnapshotUrl(this.serviceName)
+      .then((data) => {
+        this.isLoading = false;
+        this.downloadLink = data.url;
+        this.size = SnapshotDownloadController.convertSize(data.size);
+        this.downloadLinkCurlCommand = `curl "${data.url}" --output ${this.serviceName}-snapshot --fail`;
+      })
+      .catch((error) => {
+        this.isLoading = false;
+        switch (error.status) {
+          case 404:
+            this.errorMessage = `Error ${error.status}:  ${error.statusText}`;
+            break;
+          case 403:
+            this.errorMessage = `Error ${error.status}:  ${error.message}`;
+            break;
+          case 500:
+            this.errorMessage = `Error ${error.status}:  ${error.statusText}`;
+            break;
+          case 503:
+            this.errorMessage = `Error ${error.status}:  ${error.statusText}`;
+            break;
+          default:
+            this.errorMessage = `Error ${error.status}:  ${error.message}`;
+        }
+      });
+  }
+
+  isDownloadLinkAvailable() {
+    return !!this.downloadLink;
+  }
+
+  static convertSize(bytes, decimals = 2) {
+    if (!+bytes) return '0 Bytes';
+
+    const k = 1024;
+    const dm = decimals < 0 ? 0 : decimals;
+    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
```

Comment:
**question:** Does not these need translations?

Reply:
Not in the current specification.